### DISCUSSION TITLE
Roofspace power density assumption

### DISCUSSION THREAD

**Discussion by smryania** [NREL/REopt-API-Analysis#144] [2 - Site & Utility Questions]
The web tool user manual says, "PV size is constrained by ... roof area available, assuming a power density of 10 DC-Watts/ft2.".  Our consultant suggests the rooftop panels could produce 12.4 DC-Watts/ft^2.  Is there a way to change this assumption or work around it, other than by artificially increasing the roofspace?

**Comment by adfarth** [comment]
The assumed power density can be changed in the REopt Julia Package (https://nrel.github.io/REopt.jl/dev/) by modifying the `PV : kw_per_square_foot` input. However it cannot currently be modified in the web tool. 

I would suggest either artificially increasing the rooftop area available as you've mentioned, or just using the PV Maximum size (kW) input to constrain the PV to the amount that you determine can be installed at a given site.

**Comment by itsGreghere** [comment]
I found this question when searching the same issue with this excerpt from the user manual. Modern, widely available PV modules generate around 20 W/ft², so assuming a max density of 10 W/ft² for the roof is far too low in most cases. On a commercial building with a flat roof, it would be possible to get very close to 20 Watts-DC per ft² of roof area available for DER with a ~10° tilt. I would strongly recommend increasing this default density assumption to 20 in the web tool and as a default in the Julia package, if applicable.

Is it also possible to adjust `PV : kw_per_square_foot input` when using the API? I'm not finding that variable in `models.py` such as in [this class](https://github.com/NREL/REopt_API/blob/5ac66dbb5c68820ef5e2e09861f650c5051bbb36/reoptjl/models.py#L2670). Thanks in advance.

**Comment by adfarth** [comment]
@smryania the assumed rooftop and ground-mount PV density can now be changed in the web tool!

@itsGreghere thank you for this input, we will consider this! Additionally, we have just added the density inputs to the `models.py`, `PVInputs` class. You should now be able to modify these using the REopt API

**Reply by itsGreghere** [reply]
> @smryania the assumed rooftop and ground-mount PV density can now be changed in the web tool!
> 
> @itsGreghere thank you for this input, we will consider this! Additionally, we have just added the density inputs to the `models.py`, `PVInputs` class. You should now be able to modify these using the REopt API 

This is great! Thanks for the quick updates. I think they'll be appreciated by many others as well.
